csharp-intermediate-topics/LockingMechanismInCSharp/Tests/LockingMechanismInCSharpTests.cs

```diff
@@ -0,0 +1,64 @@
+using LockingMechanismInCSharp;
+
+namespace Tests
+{
+    [TestClass]
+    public class LockingMechanismInCSharpTests
+    {
+        [TestMethod]
+        public void WhenMultithreadWithoutLock_ThenDifferentResultSmallerThan3000()
+        {
+            var result = MultithreadWithoutLocking.Execute();
+
+            Assert.IsTrue(result < 300000, "Result is not smaller than 300000");
```

Comment:
I have an AMD Processor with 8 cores. Running the `Execute` in a loop of 10,000 iterations I found that the correct 300,000 result occurred 9728 times. I am not exactly sure what a better test would be. It's a hard thing to test since the failure is non-deterministic and is very hardware-dependent. Maybe running it in a loop (with a minor change in the Execute method to reset the counter to 0) until it produces the wrong result. Sorry, I don't have a better suggestion offhand, but this test as it is, will non-deterministicly cause the tests to fail.

Reply:
Unfortunately, I couldn't find a meaningful way to write this test, so I just removed it.